examples/chat/js/mutations/AddMessageMutation.js

```diff
@@ -0,0 +1,109 @@
+import Relay from 'react-relay';
+
+export default class AddMessageMutation extends Relay.Mutation {
+  static fragments = {
+    thread: () => Relay.QL`
+      fragment on Thread {
+        id,
+        isRead,
+        lastUpdated
+      }
+    `,
+    viewer: () => Relay.QL`
+      fragment on User {
+        id,
+        threads(first: 9007199254740991) {
+          unreadCount,
+          edges {
+            node {
+              id
+            }
+          }
+        },
+      }
+    `,
+  };
+  getMutation() {
+    return Relay.QL`mutation{addMessage}`;
+  }
+  getFatQuery() {
+    return Relay.QL`
+      fragment on AddMessagePayload {
+        messageEdge,
+        thread {
+          isRead,
+          lastUpdated
+        },
+        viewer {
+          threads(first: 9007199254740991) {
+            unreadCount,
+            edges {
+              node {
+                id
+              }
+            }
+          },
+        },
+      }
+    `;
+  }
+  getConfigs() {
+    return [{
+      // use FIELDS_CHANGE here to make unreadCount and thread order
+      // changeed as adding new message
+      type: 'FIELDS_CHANGE',
+      fieldIDs: {
+        thread: this.props.thread.id,
```

Comment:
`thread` might already be updated by being a parentID in the `RANGE_ADD`. Can you try with just `viewer` to see if everything is still updated?